plugins/modules/identity/sap_company.py

```diff
@@ -0,0 +1,331 @@
+        if state['TYPE'] == "S":
+            if state['NUMBER'] != '079':
+                change = True
+            else:
+                msg = "No changes where made."
+    return [{"change": change}, {"failed": failed}, {"msg": msg}]
+
+
+def run_module():
+    module = AnsibleModule(
+        argument_spec=dict(
+            state=dict(default='present', choices=['absent', 'present']),
+            conn_username=dict(type='str', required=True),
+            conn_password=dict(type='str', required=True, no_log=True),
+            host=dict(type='str', required=True),
+            sysnr=dict(type='str', default="01"),
+            client=dict(type='str', default="000"),
+            company_id=dict(type='str', required=True),
+            name=dict(type='str', required=False),
+            name_2=dict(type='str', required=False),
+            country=dict(type='str', required=False),
+            time_zone=dict(type='str', required=False),
+            city=dict(type='str', required=False),
+            post_code=dict(type='str', required=False),
+            street=dict(type='str', required=False),
+            street_no=dict(type='str', required=False),
+            e_mail=dict(type='str', required=False),
+        ),
+        supports_check_mode=False,
+    )
+    result = dict(changed=False, msg='', out={})
+    raw = ""
+
+    params = module.params
+
+    state = params['state']
+    conn_username = (params['conn_username']).upper()
+    conn_password = params['conn_password']
+    host = params['host']
+    sysnr = params['sysnr']
+    client = params['client']
+
+    company_id = (params['company_id']).upper()
+    name = params['name']
+    name_2 = params['name_2']
+    country = params['country']
+    time_zone = params['time_zone']
+    city = params['city']
+    post_code = params['post_code']
+    street = params['street']
+    street_no = params['street_no']
+    e_mail = params['e_mail']
+
+    if not HAS_PYRFC_LIBRARY:
+        module.fail_json(
+            msg=missing_required_lib('pyrfc'),
+            exception=ANOTHER_LIBRARY_IMPORT_ERROR)
+
+    # basic RFC connection with pyrfc
+    try:
+        conn = Connection(user=conn_username, passwd=conn_password, ashost=host, sysnr=sysnr, client=client)
+    except Exception as err:
+        result['error'] = str(err)
+        result['msg'] = 'Something went wrong connecting to the SAP system.'
+        module.fail_json(**result)
+
+    # build parameter dict of dict
+    company_params = build_company_params(name, name_2, country, time_zone, city, post_code, street, street_no, e_mail)
+
+    if state == "absent":
+        raw = call_rfc_method(conn, 'BAPI_COMPANY_DELETE', {'COMPANY': company_id})
+
+    if state == "present":
+        raw = call_rfc_method(conn, 'BAPI_COMPANY_CLONE',
+                              {'METHOD': {'USMETHOD': 'COMPANY_CLONE'}, 'COMPANY': company_id, 'COMP_DATA': company_params})
+
+    analysed = return_analysis(raw)
+
+    result['out'] = raw
+
```

Comment:
assignement can be done in 307 and 310, but i like it the way you chose to handle it here. It is definitely clearer that way.